pkg/sfu/buffer/buffer.go

```diff
@@ -559,6 +559,9 @@ func (b *Buffer) SetSenderReportData(rtpTime uint32, ntpTime uint64) {
 }
 
 func (b *Buffer) SetLastFractionLostReport(lost uint8) {
+	b.Lock()
+	defer b.Unlock()
+
 	b.lastFractionLostToReport = lost
```

Comment:
I know go race report that, but do we need a lock for this field?

Reply:
Not really as it is a single byte, but getting a clean report is good (I wish go race is more aware of atomic data access). As this is a lock over a very short section of code, went with it.

I would actually like to rework the locking completely in `Buffer` as it holds the lock at top level pretty much for all routines. Need to make that lock more fine grained.